analytics.js

```diff
@@ -1130,6 +1130,48 @@ class Analytics {
     processDataInAnalyticsArray(this);
   }
 
+  /**
+   * Call control pane to get client configs
+   *
+   * @param {*} writeKey
+   * @memberof Analytics
+   */
+  load(writeKey, serverUrl, options) {
+    logger.debug("inside load ");
+    // check if the below features are available in the browser or not
+    // If not present dynamically load from the polyfill cdn
+    if (
+      !String.prototype.endsWith ||
+      !String.prototype.startsWith ||
+      !String.prototype.includes ||
+      !Array.prototype.find ||
+      !Array.prototype.includes ||
+      !Promise ||
+      !Object.entries
+    ) {
+      if (!window.hasOwnProperty("polyfill")) {
+        ScriptLoader(
+          "polyfill",
+          "https://polyfill.io/v3/polyfill.min.js?features=Array.prototype.find%2CArray.prototype.includes%2CPromise%2CString.prototype.endsWith%2CString.prototype.includes%2CString.prototype.startsWith%2CObject.entries"
+        );
+      }
```

Comment:
We should improve this:
- Build a map of polyfill.io name vs feature ("Array.prototype.find": Array.prototype.find)
- If possible, move this to a different file.

Reply:
If possible, move this to a different file - you mean the whole logic or just polyfill.io name vs feature mapping?

Reply:
I meant only the mapping for now